_internals/babelfish-query-handling.md

```diff
@@ -6,43 +6,47 @@ nav_order: 2
 
 ## Babelfish query handling
 
-In this section, you will learn about Babelfish query handling. The first and
-most important thing is: Microsoft SQL Server and PostgreSQL differ quite substantially in their query handling.
-For that reason, various transformations have to be applied to make it work.
-
-Let us therefore take a closer look, and discuss some of the changes made to PostgreSQL
-core to make Babelfish work in the first place. As already stated, Babelfish is
-heavily based on protocol hooks. However, there is more to the story than that. 
+Microsoft SQL Server and PostgreSQL handle queries quite 
+differently. Babelfish utilizes protocol hooks and provides software
+mechanisms that allow you to refine and influence how your queries are processed.
 
 
 ### Processing a query
 
-The first thing to understand is how Babelfish handles a query. 
+T-SQL has many syntactic differences from the version of SQL spoken by PostgreSQL. To 
+accept queries written in T-SQL, Babelfish utilizes a parser that is distributed in 
+the `pltsql` extension. 
 
-T-SQL is a variant of SQL with a great many syntactic differences. To be able to
-accept queries written in T-SQL, Babelfish provides its own parser via the
-`pltsql` extension. Once a client has sent a query to Babelfish via the TDS
-protocol, various parser hooks are employed to parse, analyse and then finally
-execute the query.  The parser hooks create a representation suitable to be
-handed over to the `pltsql` compiler and interpreter, which finally executes the
-query.  Finally, pltsql is responsible to provide the proper datatype
-conversion from PostgreSQL types to TDS wire types. With this approach,
-processing, executing and materializing query results are located within the
-pltsql extension only, and we don't need to extend the PostgreSQL SQL parser to
-understand T-SQL and its semantics at all.
+Once a client has sent a query to Babelfish via the TDS protocol, parser hooks in the <code>pltsql</code> 
+extension are employed to parse, analyze, and execute the query. The extension provides the proper datatype 
+conversion from PostgreSQL types to TDS wire types. By taking this approach, processing, executing, 
+and materializing query results are handled by the extension only, and we don't need to extend the 
+PostgreSQL SQL parser to understand T-SQL and its semantics at all.
 
 
 ### Additional server side variables
 
-Babelfish offers new config variables to handle queries. One of the variables
-added is called <code>enable_domain_typmod</code>. To understand what this variable means,
```

Comment:
The typemod stuff is NOT for end users, it is purely for development-internal purposes, it has not been tested as a feature. Best do not document it all. If it really needs to be mentioned it needs to come without detailed explanation and with only a big warning that nobody should touch, and if they use it, database integrity may not be guaranteed.